sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -175,6 +169,17 @@ namespace Azure { namespace Core { namespace Http {
      * @return A unique pointer to an HTTP RawResponse.
      */
     virtual std::unique_ptr<RawResponse> Send(Request& request, Context const& context) override;
+
+    ~WinHttpTransport()
+    {
+      // Close the handles and set them to null to avoid multiple calls to WinHTTP to close the
+      // handles.
+      if (m_sessionHandle)
+      {
+        WinHttpCloseHandle(m_sessionHandle);
```

Comment:
A transport adapter gets initialized inside the transport policy, which gets initialized as part of a pipeline, which is the heart of an SDK client.  I am not sure if each SDK client can be OK by using one session only